### PR TITLE
Add ApiOption overloads to methods on I(Observable)OrganizationsClient

### LINKED ISSUE #1165: Add ApiOption overloads to methods on IOrganizationsClient
As part of completing #760 a number of overloads need to be added to the client interfaces.

As an example, this has been added to [`IReleasesClient`](https://github.com/octokit/octokit.net/commit/03ea83148995711fc47148489e2123e65f33bd82)and [`IObservableReleasesClient`](https://github.com/octokit/octokit.net/commit/aa3ce0db9210b778414e290af7f17c4a646bdeec).

The checklist for porting each method is:

- Add an overload for this method on `IOrganizationsClient` with a new parameter `ApiOptions`
- Implement the new method on the class which implements `IOrganizationsClient`
- Add an overload for this method on `IObservableOrganizationsClient` with a new parameter `ApiOptions`
- Implement the new method on the class which implements `IObservableOrganizationsClient`

This needs to be done for each of these methods:

- [ ] `Task<IReadOnlyList<Organization>> GetAll(String user)`
- [ ] `Task<IReadOnlyList<Organization>> GetAllForCurrent()`

### PR DESCRIPTION
**Pull request by dampir**
Fixes #1165 

To-do:

- [x]  Add overloads on IOrganizationsClient
- [x]  Add overloads on IObservableOrganizationsClient
- [x]  Add unit tests for these new methods
- [x]  Add integration tests for these new methods

    I had to use [GitHubEnterpriseTest] to get these tests done, because I should create organizations firstly.

/cc @shiftkey, @ryangribble

### CODE REVIEW

**Comment by ryangribble** [comment]
Have you got a github enterprise instance to run the integration tests against or would you like me to do that?

**Comment by dampir** [comment]
No, I haven't. Unfortunately, I don't know a lot of about enterprise integration test procedure, so do it at your discretion :smile:

**Comment by ryangribble** [comment]
> Anyway, if there is exists an possibility to provide me test enterprise credentials for period of GSoC, I think that I could run and debug enterprise integration tests by myself :+1:

[GitHub Enterprise](https://enterprise.github.com/home) is a version of GitHub that an organization hosts on their own servers (either on prem or cloud).  They do have a free 45 day trial though, so potenitally you could spin one up if you have AWS or Azure credit, or enough hardware at home for the minimum spec (16GB RAM).  But it really isn't necessary

I've pulled your stuff down and ran all the integration tests, you've just got one issue which ill highlight

File: Octokit.Tests.Integration/Clients/OrganizationClientTests.cs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                var skipStartOptions = new ApiOptions
+                {
+                    PageSize = 1,
+                    PageCount = 1,
+                    StartPage = 2
+                };
+
+                var secondPage = await _organizationsClient.GetAllForCurrent(skipStartOptions);
+
+                Assert.NotEqual(firstPage[0].Name, secondPage[0].Name);
```
Comment (ryangribble) [inline Octokit.Tests.Integration/Clients/OrganizationClientTests.cs]: `.Name` is `null` on - you should use `.Login` field and then everything is green!  :tada:

Comment (dampir) [inline Octokit.Tests.Integration/Clients/OrganizationClientTests.cs]: It's cool, especially if take into account that I wrote these tests blindly :smile: 
Fixed!

**Comment by dampir** [comment]
> GitHub Enterprise is a version of GitHub that an organization hosts on their own servers (either on prem or cloud). They do have a free 45 day trial though, so potenitally you could spin one up if you have AWS or Azure credit, or enough hardware at home for the minimum spec (16GB RAM).

Thank you for good explanation, now I really get it.

> But it really isn't necessary

If so, I really hope that there are not so many clients that should use Enterprise API (throughout my GSoC) :smile:

File: Octokit.Tests.Integration/Clients/OrganizationClientTests.cs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+                    PageCount = 1,
+                    StartPage = 2,
+                };
+
+                var organizations = await _organizationsClient.GetAllForCurrent(options);
+
+                Assert.Equal(1, organizations.Count);
+            }
+
+            [GitHubEnterpriseTest]
```
Comment (shiftkey) [inline Octokit.Tests.Integration/Clients/OrganizationClientTests.cs]: This seems reasonable. I'm not sure how we feel about creating multiple organizations for each test run - @ryangribble do you think this is worth doing simply to test pagination?

**Comment by shiftkey** [comment]
Gonna merge this in and we can come back to that open question about running the tests cc @ryangribble

**Comment by ryangribble** [comment]
I think it's ok creating the orgs, my test ghe instance already has a number of random organisations created from other tests I'd written previously... They don't really affect anything and anyone running integration tests on ghe should be using a dedicated  dev instance anyway

**Pull request merged**